Mossaka/wasmCloud-fork
# feat(example): add http blobstore example

## Feature or Problem
This PR adds an example component, `http-blobstore`, which uses `wasi:http` and `wasi:blobstore` to return the number of files in a specific filesystem folder in response to an HTTP request. The primary purpose of this example is to show a simple interaction with a blobstore to aid with testing the Azure blobstore provider 😄 

## Related Issues


## Release Information


## Consumer Impact


## Testing
You may be interested in why we implement `wasmcloud:blobstore` on the provider side but use `wasi:blobstore` for the component itself. This is something we're planning to address in a future release, supporting just `wasi:blobstore`, but for now we are polyfilling the functions so that we can send the WASI calls over our distributed RPC network. You may be interested in our `Blobstore` trait implementation to see how the `wasi:blobstore` functions map to our `wasmcloud:blobstore` operations: https://github.com/wasmCloud/wasmCloud/blob/main/crates/host/src/wasmbus/mod.rs#L377

### Unit Test(s)


### Acceptance or Integration


### Manual Verification
I did manually verify that this works with the filesystem provider, and the instructions in the example's README should work out of the box for you @Mossaka assuming you have installed `wash` 0.26.0 😄 Please let me know if you run into trouble, my main goal is to outline the example with an existing blobstore provider so testing the Azure provider can be done with the same steps.
